cmake/kconfig.cmake

```diff
@@ -30,15 +34,18 @@ set(CREATE_NEW_DOTCONFIG "")
 if(NOT EXISTS ${DOTCONFIG} OR ${BOARD_DEFCONFIG} IS_NEWER_THAN ${DOTCONFIG})
   set(CREATE_NEW_DOTCONFIG 1)
 elseif(${CONF_FILE})
-  if(${CONF_FILE} IS_NEWER_THAN ${DOTCONFIG})
-    set(CREATE_NEW_DOTCONFIG 1)
-  endif()
+  foreach(CONF_FILE_ITEM ${CONF_FILE_AS_LIST})
+    if(${CONF_FILE_ITEM} IS_NEWER_THAN ${DOTCONFIG})
+      set(CREATE_NEW_DOTCONFIG 1)
+    endif()
+  endforeach()
 endif()
 if(CREATE_NEW_DOTCONFIG)
   execute_process(
     COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/scripts/kconfig/merge_config.py -m -q
       -O ${PROJECT_BINARY_DIR}
-      ${BOARD_DEFCONFIG} ${CONF_FILE}
+      ${BOARD_DEFCONFIG}
+      ${CONF_FILE_AS_LIST}
```

Comment:
You would think so.

But CMake will actually in this context expand the list into seperate items, which is what we need.

This minimal example demonstrates the difference:

set(a "/dev/null /dev/random")
string(REPLACE " " ";" a_as_list ${a})
execute_process(COMMAND file ${a_as_list})
execute_process(COMMAND file ${a})

The first execute_process will give the program "file" two arguments and succeed,
but the second will give the program one argument and fail. e.g.

file /dev/null /dev/random
vs.
file "/dev/null /dev/random"